src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -131,9 +129,10 @@ internal class CodeWriter constructor(
    */
   fun emitModifiers(
       modifiers: Set<KModifier>,
-      implicitModifiers: Set<KModifier> = emptySet()) {
+      implicitModifiers: Set<KModifier> = emptySet()
+  ) {
     if (modifiers.isEmpty()) return
-    for (modifier in EnumSet.copyOf(modifiers)) {
+    for (modifier in modifiers.toEnumSet()) {
```

Comment:
This was being used for sorting. The new version doesn't seem to be explicitly sorting. Are we just getting lucky?

Reply:
The new implementation consumes enum values in order and picks the ones present in `modifiers` - pretty much what `EnumSet.copyOf()` does. There are a few test cases that would've caught incorrect ordering.